### ISSUE TITLE
SonarLint doesnt work with readonly projects in Connected Mode

### ISSUE THREAD

**Issue by CoffeeParser** [SonarSource/sonarlint-visualstudio#816]
### Description

Using SonarLint in Connected Mode forces every *.csproj and *.ruleset file in the solution to be modified in order to change the referenced ruleset. In cases where read only *.csproj are used this behavior makes SonarLint unusable.

### Repro steps

1. Create a C# cmd Project and add two additional new c# library projects.
2. Let sonarqube collect data for the Project with msbuild sonar runner using Default project-key
3. repeat step 2. with a new project-key
4. inside visual Studio hit SonarLint manage Connections... hit bind Default project-key
5. observe all *.csproj files have been edited and new *.ruleset files have been added to each Project inside the solution
6. repeat step 4. with the second Project-key.
7. observe that every *.ruleset file of every *.csproj file has been modified making it impossible to use SonarLint with read only projects.

### Expected behavior

SonarLint can be used with read only *.csproj Projects.

### Actual behavior

SonarLint can't be used with read only *.csproj Projects because the files cannot be modified in Scenarios where Version control marks certain Projects as readonly.

### Known workarounds

Not a Workaround but a Suggestion for a solution: 
Implement an Option to use SonarLint on solution .sln Level!
Example: Every readonly *.csproj has its `<CodeAnalysisRuleSet>HelloSonar.ruleset</CodeAnalysisRuleSet>` tag set once. Each *.ruleset file Points directly to the solution dir! e.g.
```
<?xml version="1.0" encoding="utf-8"?>
<RuleSet Name="SonarQube" ToolsVersion="15.0">
  <Include Path="$(SolutionDir)sonarqubecsharp.ruleset" Action="Default" />
</RuleSet>
```
This way only the file sonarqubecsharp.ruleset (at solution root) Needs to be updated on new bind actions! Effectively making it possible to use sonarlint for readonly Projects and rebind to other sonarqube Projects without conflicts with Version control Software!

### Related information

* SonarLint for VS Version 4.8.0.4040
* Visual Studio Version 15.7.0
* SonarQube server Version 7.6
* Languages in your solution: C#

**Comment by duncanp-sonar** [comment]
Which source control system are you using - TfVC?

> Example: Every readonly *.csproj has its `<CodeAnalysisRuleSet>HelloSonar.ruleset</CodeAnalysisRuleSet>` tag set once. Each *.ruleset file Points directly to the solution dir!

I though I'd created an issue for this at the same time I created #805, but I don't seem to have done so. I've created #817 now.

Issues #635 is also relevant.

### Workaround
Since v4.8 you can manually configure projects to directly reference the generated solution-level ruleset, and delete the project-level rulesets. SonarLint will recognize that the generated ruleset is referenced.

Prior to v4.8, SonarLint would incorrectly complain that generated ruleset was not referenced. This was fixed by #790.

**Comment by duncanp-sonar** [comment]
Effectively fixed in v7.0.
Connected Mode no longer makes changes to any project files, and does not create any files under source-controlled locations.
